nixos/modules/services/mail/dovecot.nix

```diff
@@ -7,7 +7,90 @@ let
   dovecotPkg = pkgs.dovecot;
 
   baseDir = "/run/dovecot2";
-  stateDir = "/var/lib/dovecot";
+
+  sieveToPath = text:
+    if isString text && builtins.substring 0 2 text == "/*"
+    then pkgs.writeText "script.sieve" text
+    else if types.path.check text then text
+    else pkgs.writeText "script.sieve" text;
+  scriptType = types.coercedTo types.lines sieveToPath types.path;
+
+  compileSieveScripts = pkgs.buildSieveScripts {
+    name = "dovecot2-sieveScripts";
+      plugins = cfg.sieve.plugins;
+      extensions = cfg.sieve.extensions;
+      globalExtensions = cfg.sieve.globalExtensions;
+      phases = ["buildPhase" "installPhase" "fixupPhase"];
+      buildPhase = ''
+        mkdir sieve
+        cd sieve
+        ${concatStringsSep "\n" (imap1 (i: v: ''
+          mkdir -p after/${toString i}
+          if [ -d '${v}' ]; then
+            cp -p '${v}'/*.sieve after/${toString i}
```

Comment:
Would it be possible to create symlinks instead of copying files?

Reply:
Done! And while looking at this I found that when the path was outside the store the user feedback wasn't that great and would result in an unclean build in non-sandboxed builds so I have added logic to use `fileContents` to write paths outside the store to the store first.